### DISCUSSION TITLE
How do I determine whether the current month is June according to the user's system time zone?

### DISCUSSION THREAD

**Discussion by teohhanhui** [BurntSushi/jiff#26] [Q&A] [answered]
My use case is just checking the current "local" date / time, not going to store/send it in any way and/or perform any arithmetic on it.

Specifically, the question in my case is: "Is it June now?" (as defined by the user's system)

I have been unable to find anything that would let me do that without doing the unnecessary work of looking up the IANA timezone in the tzdb, but perhaps I'm just missing the obvious?

**Comment by BurntSushi** [comment, marked as answer]
`Zoned::now().month() == 6` is exactly the correct answer to whether it is June, according to the user's system.

`time` is doing the same work. It's just farming it out to `libc`. `libc` in this case will probably open `/etc/localtime` and use that file directly to determine the time zone transitions. Jiff will do something a little smarter and look for the tzid in the symlink of `/etc/localtime`, and then go look that time zone up in `/usr/share/zoneinfo` (which is where `/etc/localtime` is linked to anyway). So Jiff isn't really doing any extra work here as far as I can tell. And the `/etc/localtime` inspection is cached.

**Reply by teohhanhui** [reply]
IIUC, this would require the bundled tzdb on Windows, right? Even though (I believe) I only need the `Bias` from [`DYNAMIC_TIME_ZONE_INFORMATION`](https://learn.microsoft.com/en-us/windows/win32/api/timezoneapi/ns-timezoneapi-dynamic_time_zone_information)

**Reply by BurntSushi** [reply]
I believe that is correct, yes. The only thing Jiff pulls from `DYNAMIC_TIME_ZONE_INFORMATION` is the `TimeZoneKeyName`.

**Reply by teohhanhui** [reply]
I see `time` uses [`SystemTimeToTzSpecificLocalTime`](https://learn.microsoft.com/en-us/windows/win32/api/timezoneapi/nf-timezoneapi-systemtimetotzspecificlocaltime): https://github.com/time-rs/time/blob/aca5c3652aff6fc7df3fbd8591c3e324d3a6ffb2/time/src/sys/local_offset_at/windows.rs#L92C17-L92C48

As it seems that helps avoid bundling tzdb on Windows, I'll stick to that for now considering my very specific use case.

But it's good to know that the design of this crate is well-considered.

**Comment by teohhanhui** [comment]
Thank you for the explanation and sorry for the noise. I realize this should have been a Q&A in the discussions instead.

**Reply by BurntSushi** [reply]
Done. I converted it. :-) Thankfully GitHub makes that easy.
